workspaces/ballerina/ballerina-extension/src/rpc-managers/ai-panel/inline-utils.ts

```diff
@@ -442,24 +447,29 @@ export async function processInlineMappings(
 ): Promise<MappingElement | ErrorCode> {
     let inlineDataMapperResponse = cleanInlineDataMapperModelResponse(request) as InlineDataMapperModelResponse;
 
-    let result = await getInlineParamDefinitions(inlineDataMapperResponse);
-    if (isErrorCode(result)) {
-        return result as ErrorCode;
-    }
-    let parameterDefinitions = (result as ParameterDefinitions).parameterMetadata;
-    if (file) {
-        let mappedResult = await mappingFileInlineDataMapperModel(file, inlineDataMapperResponse);
-        if (isErrorCode(mappedResult)) {
-            return mappedResult as ErrorCode;
+    try {
+        let result = await getInlineParamDefinitions(inlineDataMapperResponse);
+        if (isErrorCode(result)) {
```

Comment:
Ideally, we should not return an error as a union type. We should throw and use try and catch to handle the error.